.gitignore

```diff
@@ -1,125 +1,20 @@
-
-# Binaries for programs and plugins
-*.exe
-*.exe~
-*.dll
-*.so
-*.dylib
-bin
-
-# Test binary, build with `go test -c`
-*.test
-
-# Output of the go coverage tool, specifically when used with LiteIDE
-*.out
-
-# Kubernetes Generated files - skip generated files, except for vendored files
-
-!vendor/**/zz_generated.*
-
-# editor and IDE paraphernalia
-.idea
-*.swp
-*.swo
-*~
-
-# Temporary Build Files
-build/_output
-build/_test
-# Created by https://www.gitignore.io/api/go,vim,emacs,visualstudiocode
-### Emacs ###
-# -*- mode: gitignore; -*-
-*~
-\#*\#
-/.emacs.desktop
-/.emacs.desktop.lock
-*.elc
-auto-save-list
-tramp
-.\#*
-# Org-mode
-.org-id-locations
-*_archive
-# flymake-mode
-*_flymake.*
-# eshell files
-/eshell/history
-/eshell/lastdir
-# elpa packages
-/elpa/
-# reftex files
-*.rel
-# AUCTeX auto folder
-/auto/
-# cask packages
-.cask/
+vendor/
 dist/
-# Flycheck
-flycheck_*.el
-# server auth directory
-/server/
-# projectiles files
-.projectile
-projectile-bookmarks.eld
-# directory configuration
-.dir-locals.el
-# saveplace
-places
-# url cache
-url/cache/
-# cedet
-ede-projects.el
-# smex
-smex-items
-# company-statistics
-company-statistics-cache.el
-# anaconda-mode
-anaconda-mode/
-### Go ###
-# Binaries for programs and plugins
-*.exe
-*.exe~
-*.dll
-*.so
-*.dylib
-# Test binary, build with 'go test -c'
-*.test
-# Output of the go coverage tool, specifically when used with LiteIDE
 *.out
-### Vim ###
-# swap
-.sw[a-p]
-.*.sw[a-p]
-# session
+*.sw[ap]
+k8gb
+k8gb.exe
+
+# IDE specific files (debug config etc...)
+.vscode/
+.idea/
 Session.vim
-# temporary
 .netrwhist
-# auto-generated tag files
 tags
-### VisualStudioCode ###
-.vscode/*
-.history
-# End of https://www.gitignore.io/api/go,vim,emacs,visualstudiocode
 
-### JetBrains IDE ###
-.idea
-# End of idea tools
-# Exclude chart tar.gz files
+# Helm charts
 chart/k8gb/charts/*.tgz
 
-# Ignore testing kubeconfigs
-kubeconfig*
```

Comment:
Can we please leave it around? I'm using it for testing over multiple clusters from time to time avoiding tampering the main kubeconfig

Reply:
@ytsarev This looks like a user-specific, not project-specific thing :) i.e. you can `export KUBECONFIG=` to point to any `kubeconfig` path of your like, shouldn't be necessary lying in the k8gb working copy. Or am I missing something?

Reply:
@somaritane look like project creator specific thing yes ;) it is just two lines to speed up my productivity, I am not asking for too much, I think :)
Re kubeconfigs - I prefer to temporary keep them to the project under test, the matter of convenience

Reply:
@somaritane following the same logic,  you could also use vim and we could drop `.vscode` and `.idea` ;)

Reply:
Also kubeconfigs are somehow related to k8s operator project :D Sorry, now I will not give it up so easily :D

Reply:
@ytsarev I actually all for dropping IDE-specific files, they can simply reside in a personal `.gitignore` file.

Reply:
@somaritane works for me, i will sacrifice my kubeconfigs for that :D

Reply:
I am also in favor of removing IDE specific settings, I left it there more because for most users this is too conflicting.

Reply:
pls keep `.idea` & `.vscode` also there, I have both in my global `.gitignore`, but still if it's not there it makes contributing (for external ppl w/ random editors) harder. The [chances](https://insights.stackoverflow.com/survey/2021#worked-with-vs-want-to-work-with-new-collab-tools-worked-want-prof) of using either vs code or idea are pretty high these days 

I don't see anything wrong w/ putting ide specific one-liners for IGNORING the configs (like `.idea`) what seemed weird to me were those `!.vscode/tasks.json` "keep-ide-specific-stuff-in-repo" rules

btw. I had to add those two specific rules to my `.gitignore` to be able to work comfortly w/ k8gb:

```
...
/.custom-env/*               ... custom env files that can be consumed by debug/run configurations in goland
/k8gb                        ... binary file we produce when doing `go build`
```

Reply:
just a reminder, if we leave the DenyListing .gitignore and switch to AllowListing .gitignore instead, we will only allow what is actually part of the project.

Reply:
> The chances of using either vs code or idea are pretty high these days
@jkremser This statement is subjective without stats :).

I just hope that If we leave `.gitignore` for vim, idea, vscode, then we're ready to accept PRs to add emacs, sublime, atom, eclipse, komodo etc (sits pretty well in random editor category with Golang support available) on request in the future, denying such requests would be unfair. 

But as I said, if I have my IDE of choice, asking every single project to include it in its `.gitignore` won't scale.